Simon's Ground work

SimonDKnight opened the pull request:
Not finished the word count or primes yet, but I will get there.

benlovell commented on primes.rb:
```diff
@@ -1,6 +1,49 @@
+From there I know I need to them basically just check limit.each % each of the array numbers to see if they equal 0 to rule them out
+as not prime, but seem to be brain farting it atm.
+
+I am aware there is a prime gem I could use, but seems to defeat the purpose of the exercise.
```
Hey Simon! Thanks for the submission so far. Don't be disheartened, it's a tricky problem to solve for sure. I'd be happy to see you create a solution relying on the available library code initially. Why don't we try and get the submission complete this way, and then perhaps if you have the time you can come back and try and implement your own, non-library solution?

One quick point about your use of variables, prefixing your variables with `$` makes them globally scoped - which is generally something you'd want to avoid. We generally aim to limit the scope of any variables we declare and that means local or instance variables are used most often. Try and replace your own solution with library code from the `Prime` class and remember to limit the scope of your variables where possible. :+1:

benlovell commented on public/stylesheets/main.css:
```diff
@@ -1,32 +1,43 @@
 
 #content {
-  padding: 1em;
+  /*padding: 1em;*/
```
Don't be afraid to delete unwanted style declarations! If you don't need them, they're just noise. They'll always be around in the git source control repository if they need to be reinstated, or if you just need to refer to them in general.

benlovell commented on public/stylesheets/main.css:
```diff
@@ -37,3 +48,115 @@ h3 {
+}
+
+.section-six .form-group{
+
```
Pay attention to your coding style. Try and be consistent with your use of whitespace and indentation as this helps your fellow developers scan and understand your code at a glance.

benlovell commented on views/index.erb:
```diff
@@ -1,41 +1,150 @@
-    <span id="word-count">0</span> words found.
+  <div class="section-two">
+    <div class="title">
+      <h2><%=@profile.keys[0].capitalize%></h2>
```
That's a nice and simple solution to the problem, great job! However, I think we could perhaps make this more dynamic. The approach you commented out would support any number of entries (key, value) in the `@profile` hash, whereas your approach is constrained by the reliance on knowing the specific index and reaching into the `keys` property. I see some common elements and repetition here that could be perfect candidates for a more 'dynamic' approach. What do you think?

benlovell commented on views/index.erb:
```diff
@@ -1,41 +1,150 @@
+          <form action="/update" method="POST">
+            <fieldset>
+              <div class="form-group">
+                <label for="trainee[about]">About</label>
```
Great stuff. You got this working perfectly! I'll be a nag again though (if you'll excuse me :wink:) and suggest there is some opportunity to reduce some of this duplication with something more dynamic. Remembering that iterating the `@profile` hash with its `key` and `value` attributes would be a good start...

SimonDKnight commented:
Thanks for your comments Ben. I have fixed the whitespace issue and cleaned up the css file to make it more readable. 
I am going to give the primes another crack using a slightly different approach, if that does not work I will just go ahead and use the prime gem. I will also stop using all the global variables ;)

Finally I will try to make the page a little more dynamic to avoid the repetition in my code.

benlovell commented:
@SimonDKnight brilliant, do give me a shout if you need any help :+1:

SimonDKnight commented:
Hi, I have now made some extra additions to the ground work. I figured out how to get the primes working without resorting to a gem, though I am running into a very weird problem on the website (but not when run independently) where it is giving me multiples of the 1060 result from adding up the primes (will try to figure it out).

I have also dried up the form population by iterating through the profiles hash.

SimonDKnight commented:
Prime is now working 100%, managed to fix the problem by adding an initialize def where I inialize all the variable (and reset them) for each calculation.

SimonDKnight commented:
Added a solution (albeit an inefficient one) to the Project Euler prime factorisation problem.
